### PR TITLE
Set automatic lowercase conversion to the docker images

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fix https://github.com/siddhi-io/distribution/issues/584

## Goals
$subject

## Approach
Convert image name to lowercase.

## Test environment
Java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)
Google Chrome version 77.0.3865.90 (Official Build) (64-bit)

### CODE REVIEW

**Comment by pcnfernando** [comment]
Without lowercasing, can we add a validation

**Comment by BuddhiWathsala** [comment]
We can, what would be the better way?

**Comment by pcnfernando** [comment]
lets do the validation. changing the name the user provides in the impl is not correct

**Comment by niveathika** [comment]
@BuddhiWathsala Cant we do the validation at the UI level itself? IMO it will be a much better user experience

**Comment by niveathika** [comment]
Add UI validation in 9f54c9a
Screenshot from 2019-10-05 20-33-24

File: tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/EditorMicroservice.java [hunk 1]
```diff
@@ -1214,6 +1214,18 @@ public Response exportApps(
                             .entity(errorMessage)
                             .build();
                 }
+                if (!dockerBuildConfig.getImageName().equals(
+                                dockerBuildConfig.getImageName().toLowerCase())
+                ) {
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/EditorMicroservice.java]: Format

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/export-deployment-artifacts/docker-config-dialog.js [hunk 2]
```diff
@@ -61,10 +62,15 @@ define(['require', 'lodash', 'jquery'],
...
                 if (pushDocker) {
-                    if (imageName != "" && userName != "" && password != "" && email != "") {
-                        missingDockerConfigErrorMessage.hide();
+                    var upperCase = new RegExp('[A-Z]');
+                    if (!imageName.match(upperCase)) {
+                        self.dockerNameInUpperError.hide();
+                    } else {
+                        self.dockerNameInUpperError.show();
+                    }
+                    if (imageName !== "" && userName !== "" && password !== "" && email !== "") {
```
Comment (mohanvive) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/export-deployment-artifacts/docker-config-dialog.js]: Are we trim the text before above validation?

Comment (BuddhiWathsala) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/export-deployment-artifacts/docker-config-dialog.js]: Fixed in here https://github.com/siddhi-io/distribution/pull/752/commits/90bb44445c41a71bd18f5efe9b3338d5358fc112

File: tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/EditorMicroservice.java (hunk 1 above)
Comment (BuddhiWathsala) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/internal/EditorMicroservice.java]: Fixed in https://github.com/siddhi-io/distribution/pull/752/commits/9956b3889430f9b88bf867471d04b547ae4f29b9

**Pull request merged**